arch/arm/core/aarch64/reset.c

```diff
@@ -36,6 +38,28 @@ void z_arm64_el_highest_init(void)
 	isb();
 }
 
+enum el3_next_el {
+	EL3_TO_EL2,
+	EL3_TO_EL1_NO_EL2,
+	EL3_TO_EL1_SKIP_EL2
+};
+
+static inline enum el3_next_el arm64_el3_get_next_el(void)
+{
+	/*
+	 * Is considered an illegal return "[..] a return to EL2 when EL3 is
+	 * implemented and the value of the SCR_EL3.NS bit is 0 if
+	 * ARMv8.4-SecEL2 is not implemented" (D1.11.2 from ARM DDI 0487E.a)
+	 */
```

Comment:
nit: can you move this comment to the proper `EL3_TO_EL1_SKIP_EL2` case?

Reply:
Sure? I figured that the logic made more sense at the start of the thing that was checking "when EL3 is implemented and the value of the SCR_EL3.NS bit is 0 if ARMv8.4-SecEL2 is not implemented".

Reply:
Done.